app/src/main/java/com/keylesspalace/tusky/di/CoroutineScopeModule.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.di
+
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import javax.inject.Qualifier
+
+/**
+ * Scope for potentially long-running tasks that should outlive the viewmodel that
+ * started them. For example, if the API call to bookmark a status is taking a long
+ * time, that call should not be cancelled because the user has navigated away from
+ * the viewmodel that made the call.
+ *
+ * @see https://developer.android.com/topic/architecture/data-layer#make_an_operation_live_longer_than_the_screen
+ */
+@Retention(AnnotationRetention.BINARY)
+@Qualifier
+annotation class ApplicationScope
```

Comment:
I don't think I understand what this annotation does? Without it everything works just the same, no?

Reply:
It doesn't.

`AccountPreferenceDataStore.kt:16` has this line:

```
@ApplicationScope val externalScope: CoroutineScope
```

Dagger needs to know how to provide that `CoroutineScope`. But there may be many different CoroutineScopes with different dispatchers etc.

So this annotation here means that when a `@ApplicationScope ...: CoroutineScope` is injected, use this provider.

If you remove the `@ApplicationScope` annotation you get this error:

```
tusky\di\AppComponent.java:9: error: [Dagger/MissingBinding] kotlinx.coroutines.CoroutineScope cannot be provided without an @Provides-annotated method.
```

because Dagger doesn't know how to provide the correct `CoroutineScope` instance.

Reply:
No, I mean delete completely. Basically it is just here in case another injectable Scope is added in the future. Superflous if you ask me.

Reply:
From a pure code perspective you could remove the annotation and the module could just be:

```kotlin
@Module
class CoroutineScopeModule {
    @Provides
    fun providesCoroutineScope() = CoroutineScope(SupervisorJob() + Dispatchers.Default)
}
```

But the annotation provides additional context to the next person who's looking at the code, as well as a handy place to put additional explanatory Javadoc.